Assets/HoloToolkit/UX/Scripts/Lines/LineObjectCollection.cs

```diff
@@ -0,0 +1,181 @@
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Holotoolkit.Unity.UX
+{
+    public class LineObjectCollection : MonoBehaviour
+    {
+        public List<Transform> Objects = new List<Transform>();
+
+        [Range(-2f, 2f)]
+        public float DistributionOffset = 0f;
+        [Range(0f, 2f)]
+        public float LengthOffset = 0f;
+        [Range(0f, 2f)]
+        public float ScaleOffset = 0f;
+        [Range(0.001f, 2f)]
+        public float ScaleMultiplier = 1f;
+        [Range(0.001f, 2f)]
+        public float PositionMultiplier = 1f;
+
+        public float DistributionOffsetPerObject
+        {
+            get
+            {
+                return 1f / Objects.Count;
+            }
+        }
+
+        public AnimationCurve ObjectScale = AnimationCurve.Linear(0f, 1f, 1f, 1f);
+
+        public AnimationCurve ObjectPosition = AnimationCurve.Linear(0f, 0f, 1f, 0f);
+
+        public bool FlipRotation = false;
+
+        public Vector3 RotationOffset = Vector3.zero;
+
+        public Vector3 PositionOffset = Vector3.zero;
+
+        public RotationTypeEnum RotationTypeOverride = RotationTypeEnum.None;
+
+        public PointDistributionTypeEnum DistributionType = PointDistributionTypeEnum.None;
+
+        [Header("Object Placement")]
+        public StepModeEnum StepMode = StepModeEnum.Interpolated;
+
+        // Convenience functions
+        public float GetOffsetFromObjectIndex(int index, bool wrap = true)
+        {
+            if (Objects.Count == 0)
+                return 0;
+
+            if (wrap) {
+                index = WrapIndex(index, Objects.Count);
+            } else {
+                index = Mathf.Clamp(index, 0, Objects.Count - 1);
+            }
+
+            return (1f / Objects.Count * (index + 1));
+        }
+
+        public int GetNextObjectIndex(int index, bool wrap = true)
+        {
+            if (Objects.Count == 0)
+                return 0;
+
+            index++;
+
+            if (wrap) {
+                return WrapIndex(index, Objects.Count);
+            } else {
+                return Mathf.Clamp(index, 0, Objects.Count - 1);
+            }
+        }
+
+        public int GetPrevObjectIndex(int index, bool wrap = true)
+        {
+            if (Objects.Count == 0)
+                return 0;
+
+            index--;
+
+            if (wrap) {
+                return WrapIndex(index, Objects.Count);
+            } else {
+                return Mathf.Clamp(index, 0, Objects.Count - 1);
+            }
+        }
+
+        public void Update()
+        {
+            UpdateCollection();
+        }
+
+        public void UpdateCollection()
+        {
+            if (source == null)
+                source = gameObject.GetComponent<LineBase>();
+            if (source == null)
+                return;
+
+            if (transformHelper == null)
+            {
+                transformHelper = transform.Find("TransformHelper");
```

Comment:
we should serialize this `transformHelper` and give users the option to assign it in the inspector instead of searching this way.